server/service/src/ledger_fix/mod.rs

```diff
@@ -0,0 +1,87 @@
+use repository::{
+    stock_line_ledger::{StockLineLedgerFilter, StockLineLedgerRepository, StockLineLedgerRow},
+    EqualFilter, InvoiceLineFilter, InvoiceLineRepository, InvoiceLineType, InvoiceStatus,
+    RepositoryError, StockLineRowRepository, StorageConnection,
+};
+
+pub mod find_ledger_discrepancies;
+mod fixes;
+pub mod ledger_fix_driver;
+pub mod stock_line_ledger_fix;
+
+pub(crate) fn is_ledger_fixed(
+    connection: &StorageConnection,
+    stock_line_id: &str,
+) -> Result<bool, RepositoryError> {
+    let ledger_lines = StockLineLedgerRepository::new(connection).query_by_filter(
+        StockLineLedgerFilter::new().stock_line_id(EqualFilter::equal_to(stock_line_id)),
+    )?;
+
+    if ledger_lines.iter().any(|line| line.running_balance < 0.0) {
+        return Ok(false);
+    }
+
+    let balance_summary = ledger_balance_summary(connection, &ledger_lines, stock_line_id)?;
+
+    return Ok(balance_summary.is_fixed);
+}
+
+#[derive(Debug)]
+pub(crate) struct LedgerBalanceSummary {
+    is_fixed: bool,
+    available: f64,
+    total: f64,
+    running_balance: f64,
+    reserved_not_picked: f64,
+}
+
+pub(crate) fn ledger_balance_summary(
```

Comment:
nit - could be `LedgerBalanceSummary::new(...)`

Reply:
With "new" we loose sense of order of f64 fields (since they would all just be params separated by ',' easy to make mistake), i typically think new is not very useful in most cases

Reply:
I don't follow isn't it just:
```rs
let balance_summary = ledger_balance_summary(connection, &ledger_lines, stock_line_id)?;
// vs 
let balance_summary = LedgerBalanceSummary::new(connection, &ledger_lines, stock_line_id)?;
```
Note this truly doesn't matter just trying to figure out why sometimes we favour one over another!